lib/attrsets.nix

```diff
@@ -1209,6 +1246,41 @@ rec {
     in
     recurse [ ] set;
 
+  /**
+    Like [`lib.attrsets.mapAttrsRecursiveCond`](#function-library-lib.attrsets.mapAttrsRecursiveCond) but also recurses on lists. In other words it recurses over an arbitrary nesting of attrsets and lists.
+
+    :::{#map-data-recursive-cond-example .example}
+    # Map over leaf attributes defined by a condition
+
+    Map derivations to their `name` attribute.
+    Derivatons are identified as attribute sets that contain `{ type = "derivation"; }`.
+    ```nix
+    mapDataRecursiveCond
+      (as: !(as ? "type" && as.type == "derivation"))
+      (x: x.name)
+      data
```

Comment:
A complete example, including a value for `data` and the evaluation result would be helpful.